packages/core/src/api/profile.ts

```diff
@@ -0,0 +1,65 @@
+/**
+ * Copyright (c) 2024, WSO2 LLC. (https://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import {AuthClient} from 'src/auth-client';
+import AsgardeoUIException from 'src/exception';
+import {MeAPIResponse} from 'src/models/me-api-response';
+
+const getProfileInformation = async (): Promise<MeAPIResponse> => {
+  let baseUrl: string;
+  let accessToken: string;
+  let response: Response;
+
+  try {
+    baseUrl = (await AuthClient.getInstance().getDataLayer().getConfigData()).baseUrl;
+    accessToken = await AuthClient.getInstance().getAccessToken();
+  } catch (error) {
+    throw new AsgardeoUIException(
+      'JS_UI_CORE-ME-M-NF',
+      'Failed in getting the base URL and access token.',
+      error.stack,
+    );
+  }
+
+  if (!accessToken) {
+    throw new AsgardeoUIException('JS_UI_CORE-ME-M-IV', 'Access token is null.');
+  }
```

Comment:
Shall we extract this logic to a reusable function? Retrieving the access token is going to be reused by many methods I guess.

Ideally, it should be the end users responsibility to pass the correct payload and headers to `getProfileInformation` function. The SDK should not make token requests for each API call, IMO.

Reply:
In the core SDK, only this function is getting the access token. So retrieving access token won't be reused.

End user will be accessing either a custom hook to get user information or will use the profile component directly. In these scenarios we cannot expect the the end user to pass any payload or headers, as it will degrade the user experience.